--- origsrc/xcb-util-image-0.3.9/image/xcb_image.c	2011-06-30 03:05:22.000000000 -0500
+++ src/xcb-util-image-0.3.9/image/xcb_image.c	2014-04-04 16:54:19.959203900 -0500
@@ -187,6 +187,7 @@ xcb_image_create_native (xcb_connection_
       assert(0);
   }
   assert(0);
+  return NULL;
 }
 
 
@@ -751,6 +751,7 @@ xcb_image_get_pixel (xcb_image_t *image,
   default:
       assert(0);
   }
+  return 0;
 }
 
 
