--- origsrc/xcb-util-image-0.3.9/configure.ac	2014-04-04 18:48:47.019404500 -0500
+++ src/xcb-util-image-0.3.9/configure.ac	2014-04-04 18:48:34.550691300 -0500
@@ -7,12 +7,20 @@ AM_MAINTAINER_MODE
 
 XCB_UTIL_COMMON([1.4], [1.6])
 
-AC_CHECK_HEADERS([sys/shm.h])
-AM_CONDITIONAL(HAVE_SHM, test x$ac_cv_header_sys_shm_h = xyes)
-PKG_CHECK_MODULES(XCB_SHM, xcb-shm)
 PKG_CHECK_MODULES(XPROTO, xproto >= 7.0.8)
 PKG_CHECK_MODULES(XCB_UTIL, xcb-util)
 
+PC_REQUIRES="xcb"
+HAVE_SHM=no
+AC_CHECK_HEADER([sys/shm.h],
+                [PKG_CHECK_MODULES(XCB_SHM, xcb-shm,[
+                    AC_DEFINE(HAVE_SHM, 1, [Enable xcb-shm code])
+                    PC_REQUIRES="$PC_REQUIRES xcb-shm"
+                    HAVE_SHM=yes])
+                ])
+AC_SUBST(PC_REQUIRES)
+AM_CONDITIONAL(HAVE_SHM, test x$HAVE_SHM = xyes)
+
 AC_OUTPUT([Makefile
 	image/Makefile image/xcb-image.pc
 	xcb_util_intro
--- origsrc/xcb-util-image-0.3.9/image/xcb-image.pc.in	2011-04-13 21:44:37.000000000 -0500
+++ src/xcb-util-image-0.3.9/image/xcb-image.pc.in	2014-04-04 18:47:34.189238800 -0500
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: XCB Image library
 Description: XCB image convenience library
 Version: @PACKAGE_VERSION@
-Requires: xcb xcb-shm
+Requires: @PC_REQUIRES@
 Libs: -L${libdir} -lxcb-image @LIBS@
 Cflags: -I${includedir}
--- origsrc/xcb-util-image-0.3.9/image/xcb_image.c	2014-04-04 18:48:47.023404700 -0500
+++ src/xcb-util-image-0.3.9/image/xcb_image.c	2014-04-04 18:47:05.746612000 -0500
@@ -28,7 +28,9 @@
 #include <string.h>
 
 #include <xcb/xcb.h>
+#ifdef HAVE_SHM
 #include <xcb/shm.h>
+#endif
 #include <xcb/xcb_aux.h>
 #include "xcb_bitops.h"
 #include "xcb_image.h"
@@ -463,6 +465,7 @@ xcb_image_put (xcb_connection_t *  conn,
 
 
 
+#ifdef HAVE_SHM
 /*
  * Shm stuff
  */
@@ -529,6 +532,7 @@ xcb_image_shm_get (xcb_connection_t *
       return 1;
   }
 }
+#endif /* HAVE_SHM */
 
 
 static uint32_t
