--- origsrc/xcb-util-image-0.3.9/configure.ac	2012-05-30 00:22:52.000000000 -0500
+++ src/xcb-util-image-0.3.9/configure.ac	2014-04-04 17:08:32.536084100 -0500
@@ -7,12 +7,17 @@ AM_MAINTAINER_MODE
 
 XCB_UTIL_COMMON([1.4], [1.6])
 
-AC_CHECK_HEADERS([sys/shm.h])
-AM_CONDITIONAL(HAVE_SHM, test x$ac_cv_header_sys_shm_h = xyes)
-PKG_CHECK_MODULES(XCB_SHM, xcb-shm)
 PKG_CHECK_MODULES(XPROTO, xproto >= 7.0.8)
 PKG_CHECK_MODULES(XCB_UTIL, xcb-util)
 
+HAVE_SHM=no
+AC_CHECK_HEADER([sys/shm.h],
+                [PKG_CHECK_MODULES(XCB_SHM, xcb-shm,[
+                    AC_DEFINE(HAVE_SHM, 1, [Enable xcb-shm code])
+                    HAVE_SHM=yes])
+                ])
+AM_CONDITIONAL(HAVE_SHM, test x$HAVE_SHM = xyes)
+
 AC_OUTPUT([Makefile
 	image/Makefile image/xcb-image.pc
 	xcb_util_intro
--- origsrc/xcb-util-image-0.3.9/image/xcb_image.c	2011-06-30 03:05:22.000000000 -0500
+++ src/xcb-util-image-0.3.9/image/xcb_image.c	2014-04-04 16:54:19.959203900 -0500
@@ -28,7 +28,9 @@
 #include <string.h>
 
 #include <xcb/xcb.h>
+#ifdef HAVE_SHM
 #include <xcb/shm.h>
+#endif
 #include <xcb/xcb_aux.h>
 #include "xcb_bitops.h"
 #include "xcb_image.h"
@@ -462,6 +464,7 @@ xcb_image_put (xcb_connection_t *  conn,
 
 
 
+#ifdef HAVE_SHM
 /*
  * Shm stuff
  */
@@ -528,6 +531,7 @@ xcb_image_shm_get (xcb_connection_t *
       return 1;
   }
 }
+#endif /* HAVE_SHM */
 
 
 static uint32_t
